### PR TITLE
feat: Add account scanner terraform deployment

### PR DESCRIPTION
**Pull request by felipecosta09**
# Add account scanner terraform deployment

## Change Summary
Add account scanner terraform deployment and updates to the README.

## PR Checklist

- [ ] I've read and followed the [Contributing Guide](https://github.com/trendmicro/cloudone-filestorage-plugins/blob/master/.github/CONTRIBUTING.md).
- Documents/Readmes
  - [X] Updated accordingly
  - [ ] Not required
- Plugins that have versioning
  - [ ] Version bumped and follows [Semantic Versioning](https://semver.org/)
  - [X] Not required

### CODE REVIEW

**Comment by jack-c-tang** [comment]
Hi @trend-eric-pj-wang , please help review the PR. Thanks.

File: deployment/aws-terraform/README.md [hunk 1]
```diff
@@ -38,3 +38,14 @@ Deploy File Storage Security stacks across different cloud providers with Terraf
...
+## Deploy Account Scanner Stack
+
+- Deployment
+
+    1. `terraform -chdir=account-scanner init`
+    2. `terraform -chdir=account-scanner plan -var="AWSRegion=TheRegionOfDeployment"  -var="ExternalID=YourExternalID"`
+    3. `terraform -chdir=account-scanner apply -var="AWSRegion=TheRegionOfDeployment"  -var="ExternalID=YourExternalID"`
+
+- Clean Up
+    - `terraform -chdir=account-scanner destroy -var="AWSRegion=TheRegionOfDeployment"`
```
Comment (trend-eric-pj-wang) [inline deployment/aws-terraform/README.md:51]:
```suggestion
    1. `terraform -chdir=account-scanner-stack init`
    2. `terraform -chdir=account-scanner-stack plan -var="AWSRegion=TheRegionOfDeployment" -var="ExternalID=YourExternalID"`
    3. `terraform -chdir=account-scanner-stack apply -var="AWSRegion=TheRegionOfDeployment" -var="ExternalID=YourExternalID"`

- Clean Up
    - `terraform -chdir=account-scanner-stack destroy -var="AWSRegion=TheRegionOfDeployment"`
```

Follow the name of directory for storing account-scanner-stack scripts. The working directory of terraform should be `account-scanner-stack`.

File: deployment/aws-terraform/account-scanner-stack/main.tf [hunk 2]
```diff
@@ -0,0 +1,23 @@
+resource "aws_cloudformation_stack" "fss-account-scanner-by-tf" {
+  name = "fss-account-scanner-by-tf"
```
Comment (trend-eric-pj-wang) [inline deployment/aws-terraform/account-scanner-stack/main.tf:2]: For more flexible, you could add a variable such as `CFNStackName` in variables.tf and set a default value `fss-account-scanner-by-tf` for user to to define the CloudFormation stack's name.

File: deployment/aws-terraform/account-scanner-stack/main.tf [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    QuarantineBucket = var.QuarantineBucket
+    FSSBucketName = var.FSSBucketName
+    FSSKeyPrefix = var.FSSKeyPrefix
+    TrendMicroManagementAccount = var.TrendMicroManagementAccount
+    CloudOneRegion = var.CloudOneRegion
+    ExternalID = var.ExternalID
+    EnableCrossAccountScanning = var.EnableCrossAccountScanning
+  }
+
+  template_url="https://file-storage-security.s3.amazonaws.com/latest/templates/FSS-Account-Scanner-Stack.template"
```
Comment (trend-eric-pj-wang) [inline deployment/aws-terraform/account-scanner-stack/main.tf:21]: For more flexible, you could add a variable such as `TemplateURL` in variables.tf and set a default value `https://file-storage-security.s3.amazonaws.com/latest/templates/FSS-Account-Scanner-Stack.template`.

File: deployment/aws-terraform/account-scanner-stack/variables.tf [hunk 4]
```diff
@@ -0,0 +1,83 @@
+variable "AWSRegion" {
+  type = string
+  default = ""
+  description = "The AWS Region that the Stack will run"
```
Comment (trend-eric-pj-wang) [inline deployment/aws-terraform/account-scanner-stack/variables.tf:4]:
```suggestion
  description = "The AWS Region that the stack will run."
```

File: deployment/aws-terraform/account-scanner-stack/variables.tf [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+variable "ScannerEphemeralStorage" {
+  type = number
+  default = 512
+  description = "The size of the scanner lambda function's temp directory in MB. The default value is 512, but it can be any whole number between 512 and 2048 MB. Configure a large ephemeral storage to scan larger files in zip files. For more information, see https://docs.aws.amazon.com/lambda/latest/dg/configuration-function-common.html#configuration-ephemeral-storage"
+}
+
+variable "QuarantineBucket" {
+  type = string
+  default = ""
+  description = "[Optional] The bucket to quarantine malicious files. The bucket region should be the same region as the account scanner stack. Leave the bucket blank to disable quarantining"
```
Comment (trend-eric-pj-wang) [inline deployment/aws-terraform/account-scanner-stack/variables.tf:46]:
```suggestion
  description = "[Optional] The bucket to quarantine malicious files. The bucket region should be the same region as the account scanner stack. Leave it blank to disable quarantining"
```

File: deployment/aws-terraform/account-scanner-stack/variables.tf [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+variable "FSSKeyPrefix" {
+  type = string
+  default = "latest/"
+  description = "File Storage Security key prefix can include numbers, lowercase letters, uppercase letters, hyphens (-), and forward slash (/)."
+}
+
+variable "TrendMicroManagementAccount" {
+  type = number
```
Comment (trend-eric-pj-wang) [inline deployment/aws-terraform/account-scanner-stack/variables.tf:62]:
```suggestion
  type = string
```
The account ID has the possibility of beginning with 0.

File: deployment/aws-terraform/account-scanner-stack/main.tf (hunk 2 above)
Comment (felipecosta09) [reply deployment/aws-terraform/account-scanner-stack/main.tf:2]: make sense!

File: deployment/aws-terraform/account-scanner-stack/main.tf (hunk 3 above)
Comment (felipecosta09) [reply deployment/aws-terraform/account-scanner-stack/main.tf:21]: Make sense!

**Comment by felipecosta09** [comment]
all done @trend-eric-pj-wang! Thanks for the suggestions!

File: deployment/aws-terraform/account-scanner-stack/variables.tf [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+variable "KMSKeyARNForDLQSSE" {
+  type = string
+  default = ""
+  description = "The ARN for the KMS master key used to encrypt messages in SQS. Leave it blank if you haven't used your own CMK for SQS server-side encryption."
+}
+
+variable "KMSKeyARNForTopicSSE" {
+  type = string
+  default = ""
+  description = "The ARN for the KMS master keys used to encrypt messages in SNS scanResultTopic. Leave it blank if you haven't used your own CMK for SNS server-side encryption."
```
Comment (trend-eric-pj-wang) [inline deployment/aws-terraform/account-scanner-stack/variables.tf:28]:
```suggestion
  description = "The ARN for the KMS master key used to encrypt messages in SNS scanResultTopic. Leave it blank if you haven't used your own CMK for SNS server-side encryption."
```

File: deployment/aws-terraform/account-scanner-stack/variables.tf [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+variable "CFNStackName" {
+  type = string
+  default = "fss-account-scanner-by-tf"
+  description = "CloudFormation stack name."
+}
+
+variable "TemplateURL" {
+  type = string
+  default = "https://file-storage-security.s3.amazonaws.com/latest/templates/FSS-Account-Scanner-Stack.template"
+  description = "The URL of the CloudFormation template."
```
Comment (trend-eric-pj-wang) [inline deployment/aws-terraform/account-scanner-stack/variables.tf:16]:
```suggestion
  description = "The S3 URL of the account scanner CloudFormation template."
```

**Comment by trend-eric-pj-wang** [comment]
@jack-c-tang The code LGTM.

Confirm that, these tf scripts could be used to create/ update account scanner stack and remove account scanner stack.

**Review by jack-c-tang — COMMENTED** [review]
Reviewed. Please insert a changeline in the end of each file.

**Comment by felipecosta09** [comment]
> Reviewed. Please insert a changeline in the end of each file.

Done!

**Review by jack-c-tang — APPROVED** [review]

**Pull request merged**